File: python/grains/grains.py
def valid_number(f):
    def wrapper(number):
        if number > 64 or number < 1:
            raise ValueError("square must be between 1 and 64")
        return f(number)
    return wrapper


@valid_number
def square(number):
    return 1 << (number - 1)


def total():
    return (1 << 64) - 1

File: javascript/grains/grains.js
/**
 * You can use the bigint type and BigInt global object to support numbers below
 * Number.MIN_SAFE_INTEGER and above NUMBER.MAX_SAFE_INTEGER.
 *
 * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
 */
export const square = (n) => {
  if (n < 1 || n > 64)
    throw new Error('square must be between 1 and 64');

  return 2n ** BigInt(n - 1);
}

export const total = () => 2n ** BigInt(64) - 1n;
